source/server/hot_restarting_child.cc

```diff
@@ -172,5 +205,37 @@ void HotRestartingChild::mergeParentStats(Stats::Store& stats_store,
   stat_merger_->mergeStats(stats_proto.counter_deltas(), stats_proto.gauges(), dynamics);
 }
 
+void HotRestartingChild::onSocketEventUdpForwarding() {
+  std::unique_ptr<HotRestartMessage> wrapped_request;
+  while ((wrapped_request =
+              udp_forwarding_rpc_stream_.receiveHotRestartMessage(RpcStream::Blocking::No))) {
+    if (wrapped_request->requestreply_case() == HotRestartMessage::kReply) {
+      ENVOY_LOG(
+          error,
+          "HotRestartMessage reply received on UdpForwarding (we want only requests); ignoring.");
+      continue;
+    }
+    switch (wrapped_request->request().request_case()) {
+    case HotRestartMessage::Request::kForwardedUdpPacket: {
+      const auto& req = wrapped_request->request().forwarded_udp_packet();
+      Network::UdpRecvData data;
+      data.addresses_.local_ = Network::Utility::parseInternetAddressAndPort(req.local_addr());
```

Comment:
What if the listener was on a unix-domain socket (and so the address isn't an internet address)?

Reply:
Maybe? Not sure. You can have a datagram UDS on unix.

Reply:
Do you have a suggestion, is there a better helper function I could call that would parse out all the kinds of address?